### DISCUSSION TITLE
Running tests and enabling cheats outside forge command

### DISCUSSION THREAD

**Discussion by miohtama** [foundry-rs/foundry#4475] [Q&A] [answered]
I am attempting to run forge-based test contracts (or scripts) from Python + Anvil.

Do I need to do something special to set up the cheats, enable `vm_prank` and others?

**Comment by mds1** [comment, marked as answer]
Cheat codes and the `Vm` interface are only usable in forge, you can do the equivalent in anvil with RPC calls https://book.getfoundry.sh/reference/anvil/#custom-methods

**Comment by miohtama** [comment]
What it would take to enable cheat codes outside forge if someone wants to run `Vm` interface compatible scripts outside forge?

**Reply by mds1** [reply]
You'd have to identify that the script is running against anvil and convert calls to the cheatcode to RPC calls to the node. Need to also make sure the user actually wants to do that conversion, and only support the conversion during script execution, not during tests

An alternative solution would be a `vm.rpc` cheat. That with the existing JSON helpers might make it pretty easy to call the anvil RPC methods from a forge script: https://github.com/foundry-rs/foundry/issues/3362

**Reply by miohtama** [reply]
Thank you so much @mds1 

How/where Forge VM -> cheat bridge is currently set up if I want to do it myself?

I am happy to take a peak at Rust source code and learn more.

**Reply by mds1** [reply]
This should be a good starting point: https://github.com/foundry-rs/foundry/blob/master/docs/dev/cheatcodes.md
